website/src/components/Visualisations/IIGOPayments/Util/ProcessedPaymentsData.ts

```diff
@@ -0,0 +1,84 @@
+import { OutputJSONType, TeamName } from '../../../../consts/types'
+import { ProcessedTaxData, TaxData } from './IIGOPaymentsTypes'
+
+export const processPaymentsData = (data: OutputJSONType): ProcessedTaxData => {
+    if (data.GameStates.length === 0) return []
+
+    const retData: ProcessedTaxData = []
+    const IIGOHistory = Object.entries(
+        data.GameStates[data.GameStates.length - 1].IIGOHistory
+    )
+
+    const empty = (): TaxData => {
+        return {
+            expectedAlloc: 0,
+            actualAlloc: 0,
+            expectedTax: 0,
+            actualTax: 0,
+            expectedSanction: 0,
+            actualSanction: 0,
+        }
+    }
+
+    const TaxInfo: Map<TeamName, TaxData> = new Map<TeamName, TaxData>([
+        [1, empty()],
+        [2, empty()],
+        [3, empty()],
+        [4, empty()],
+        [5, empty()],
+        [6, empty()],
+    ])
+
+    IIGOHistory.forEach(([_, exchanges]) => {
+        if (exchanges) {
+            exchanges.forEach((teamAction) => {
+                const team = TeamName[teamAction.ClientID]
+                const current: TaxData | undefined = TaxInfo.get(team)
+                const type = teamAction.Pairs[0].VariableName
+                switch (type) {
+                    case 'IslandAllocation':
+                        if (current) {
```

Comment:
seems like every branch of the cases do `if (current)`, why not do it above L38 once :)